src/librustc_mir_build/hair/pattern/const_to_pat.rs

```diff
@@ -106,6 +106,13 @@ impl<'a, 'tcx> ConstToPat<'a, 'tcx> {
                     traits::NonStructuralMatchTy::Param => {
                         bug!("use of constant whose type is a parameter inside a pattern")
                     }
+                    traits::NonStructuralMatchTy::Dynamic => {
+                        // span_fatal avoids ICE in hair.
+                        self.tcx()
+                            .sess
+                            .struct_span_err(self.span, "trait object cannot be used in patterns").emit();
+                        return inlined_const_as_pat;
```

Comment:
This is not what I suggested. What I was pointing to was this: https://github.com/rust-lang/rust/blob/458bae55e7130e817dfa4e9380848c200c1e8e99/src/librustc_mir_build/hair/pattern/const_to_pat.rs#L104-L122

That `msg` variable is then used instead of `make_msg()` below.

Reply:
it doesn't change the returned pat though. Still causes the same ICE

`error: internal compiler error: src/librustc_trait_selection/traits/codegen/mod.rs:61: Encountered error `Unimplemented` selecting `Binder(<dyn std::cmp::PartialEq<u32> as std::cmp::PartialEq>)` during codegen`